### PR TITLE
Document bonded potentials

### LINKED ISSUE #3049: Bonded potentials lacking documentation
* https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/python/espressomd/interactions.pyx#L2529
* https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/python/espressomd/interactions.pyx#L2777
* https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/python/espressomd/interactions.pyx#L2859
* https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/python/espressomd/interactions.pyx#L2905
* https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/python/espressomd/interactions.pyx#L2951
* https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/python/espressomd/interactions.pyx#L3149
* https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/python/espressomd/interactions.pyx#L3195
* https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/python/espressomd/interactions.pyx#L3250

### PR DESCRIPTION
**Pull request by jngrad**
Fixes #3049

Description of changes:
- bonded IAs listed in #3049 are now documented in both Sphinx and Doxygen
- cleaned up outdated docstrings
- re-shuffled paragraphs in [7. Bonded interactions](http://espressomd.org/html/doc/inter_bonded.html)

### CODE REVIEW

File: doc/sphinx/inter_bonded.rst [hunk 1]
```diff
@@ -262,6 +284,177 @@ displayed in the visualization.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Dihedral interactions are available through the :class:`espressomd.interactions.Dihedral` class::
+
+    from espressomd.interactions import Dihedral
+    dihedral = Dihedral(bend=<K>, mult=<n>, phase=<phi_0>)
+    system.bonded_inter.add(dihedral)
+    system.part[1].add_bond((dihedral, 0, 2, 3))
+
+This creates a bond type with identifier with a dihedral potential, a
```
Comment (KaiSzuttor) [inline doc/sphinx/inter_bonded.rst]: something seems to be missing here...

**Review by KaiSzuttor — COMMENTED** [review]
really nice, thank you @jngrad just correct the one sentence and we can merge

File: doc/sphinx/inter_bonded.rst (hunk 1 above)
Comment (jngrad) [inline doc/sphinx/inter_bonded.rst]:
```suggestion
This creates a bond type identifier with a dihedral potential, a
```

Comment (jngrad) [reply doc/sphinx/inter_bonded.rst]: fixed in 507809d

**Review by KaiSzuttor — APPROVED** [review]

**Comment by jngrad** [comment]
bors r=KaiSzuttor

**Comment by jngrad** [comment]
bors retry

**Comment by jngrad** [comment]
bors r=KaiSzuttor

**Pull request merged**